test/jdk/javax/swing/ToolTipManager/bug5078214.java

```diff
@@ -41,46 +41,31 @@
 import java.awt.Rectangle;
 import java.awt.Toolkit;
 import java.awt.Window;
+import java.awt.event.MouseEvent;
 import javax.swing.JButton;
 import javax.swing.JFrame;
 import javax.swing.SwingUtilities;
 import javax.swing.ToolTipManager;
 
+import jtreg.SkippedException;
+
 public class bug5078214 {
     static volatile boolean passed = false;
 
     static volatile JFrame mainFrame;
     static volatile Rectangle bounds;
     static volatile Insets insets;
-    static volatile JRobot r;
+    static JRobot robot;
 
     public static void main(String[] args) throws Exception {
         try {
-            SwingUtilities.invokeAndWait(() -> {
-                GraphicsEnvironment ge = GraphicsEnvironment.
-                        getLocalGraphicsEnvironment();
-                GraphicsDevice[] gs = ge.getScreenDevices();
-                GraphicsConfiguration testGraphics = null;
-                for (int j = 0; j < gs.length; j++) {
-                    GraphicsDevice gd = gs[j];
-                    GraphicsConfiguration[] gc =
-                            gd.getConfigurations();
-                    for (int i = 0; i < gc.length; i++) {
-                        insets = Toolkit.getDefaultToolkit().getScreenInsets(gc[i]);
-                        if (insets.bottom != 0) {
-                            testGraphics = gc[i];
-                            break;
-                        }
-                    }
-                }
-
-                if (testGraphics == null) {
-                    System.out.print("We need at least one screen with ");
-                    System.out.println("the taskbar at the bottom position.");
-                    System.out.println("Testing skipped.");
-                    return;
-                }
+            bounds = getGraphicsConfig().getBounds();
+            if (bounds == null) {
+                throw new SkippedException("We need at least one screen " +
+                        "with the taskbar at the bottom position.");
+            }
```

Comment:
This code will never throw `SkippedException`.

First you have to test whether `getGraphicsConfig()` returned `null` and throw the exception. Otherwise, `NullPointerException` will be thrown, which fails the test instead of skipping it.

Alternatively, you may throw `SkippedException` from `getGraphicsConfig()` instead of returning `null`.